catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -497,7 +492,13 @@ private void buildQueue(Stream<Path> ingestStream, Queue<Metacard> metacardQueue
                             }
 
                             if (result != null) {
-                                metacardQueue.add(result);
+                                try {
+                                    metacardQueue.put(result);
```

Comment:
Can it be interrupted? If so then whennits interrupted you lose that ingest card. You might need to add logic to check return value of put and re try